python/cudf/cudf/core/series.py

```diff
@@ -1215,22 +1216,23 @@ def _prep_for_binop(
         can_reindex: bool = False,
         *args,
         **kwargs,
-    ):
+    ) -> Tuple[
+        Union[
+            Dict[Optional[str], Tuple[ColumnBase, Any, bool, Any]],
+            Type[NotImplemented],
+        ],
+        Optional[BaseIndex],
+    ]:
         # Specialize binops to align indices.
-        if isinstance(other, SingleColumnFrame):
+        if isinstance(other, Series):
             if (
-                # TODO: The can_reindex logic also needs to be applied for
-                # DataFrame (the methods that need it just don't exist yet).
                 not can_reindex
                 and fn in cudf.utils.utils._EQUALITY_OPS
-                and (
-                    isinstance(other, Series)
-                    # TODO: mypy doesn't like this line because the index
-                    # property is not defined on SingleColumnFrame (or Index,
-                    # for that matter). Ignoring is the easy solution for now,
-                    # a cleaner fix requires reworking the type hierarchy.
-                    and not self.index.equals(other.index)  # type: ignore
-                )
+                # TODO: mypy doesn't like this line because the index property
+                # is not defined on SingleColumnFrame (or Index, for that
+                # matter). Ignoring is the easy solution for now, a cleaner fix
+                # requires reworking the type hierarchy.
+                and not self.index.equals(other.index)  # type: ignore
```

Comment:
can we now remove mypy type ignore since it is guaranteed to be a Series?

Reply:
Yes! Great call.